include/pstl_offload/internal/usm_memory_replacement.h

```diff
@@ -184,19 +204,29 @@ __internal_operator_new(std::size_t __size, std::size_t __alignment, const std::
     return __res;
 }
 
-} // namespace __pstl_offload
-
-#if __linux__
+static bool
+__verify_aligned_new_param(std::size_t __alignment)
+{
+    if (!__is_power_of_two(__alignment))
+    {
+#if _WIN64
+        errno = EINVAL;
```

Comment:
It seems that we have this power of 2 check, but could probably generalize it to be used in the other places we check power of 2, and set `EINVAL` if not.

Reply:
Thanks, I tried to use `__verify_aligned_new_param()` when it's appropriate.